include/Mouse.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef MOUSE_HPP
+# define MOUSE_HPP
+
+# include <iostream>
+# include <OISMouse.h>
+# include <Overlay/OgreOverlay.h>
+# include <Overlay/OgreOverlayManager.h>
+# include "Input.hpp"
+
+class Mouse : public Input<OIS::Mouse, OIS::MouseListener> {
+
+	public:
+		Mouse(Mouse const &) = delete;
+		Mouse(Mouse &&) = delete;
+		virtual ~Mouse(void) = default;
```

Comment:
Ah sorry, I didn't notice you were inheriting from `Input<...>`
Yes virtual dtor is necessary in that case.